### DISCUSSION TITLE
error: no stream matches subject

### DISCUSSION THREAD

**Discussion by ondbyte** [nats-io/nats.go#1178] [Q&A] [answered]
Hi,
Can anybody help me with this
```go
//able to connect to the server
//then
//I add a stream
info,err:=js.AddStream(&nats.StreamConfig{Name: "SOMESTREAM"})
//err is nil
//When i do
_,err=js.Subscribe("SOMESTREAM.something",func(msg *nats.Msg) {
                ...
})
//here I'm receiving error "no stream matches subject"
//even
name, err := js.StreamNameBySubject("SOMESTREAM.something")
//gives the same error
```
why I'm getting this error "no stream matches subject"

using:
nats.go v1.22.1

**Comment by Jarema** [comment, marked as answer]
It's because your stream accepts literal `SOMESTREAM` subject, not `SOMESTREAM.something`.

To have it working do:
```
info,err:=js.AddStream(&nats.StreamConfig{Name: "SOMESTREAM", subjects: []string{"SOMESTREAM.*"})
```

you can also use `SOMESTREAM.>` which will match every token after `SOMESTREAM`, i.e. `SOMESTREAM.data.other`.

**Reply by ondbyte** [reply]
thanks for the answer, much appreciated, its working now
